lib/util.js

```diff
@@ -573,7 +585,8 @@ exports.isPrimitive = isPrimitive;
 function isBuffer(arg) {
   return arg instanceof Buffer;
 }
-exports.isBuffer = isBuffer;
+exports.isBuffer = deprecate(isBuffer,
+    'util.isBuffer is deprecated, please use a user-land alternative.');
```

Comment:
> I'm not sure why Buffer.isBuffer was added initially. That'd be interesting to find out.

Once upon a time there existed the SlowBuffer class.  It didn't inherit from Buffer so instances were buffers but not `instanceof Buffer`.

SlowBuffer still exists but it's more or less obsolete.  It inherits from Buffer now so `instanceof` checks work.

As to why it didn't before, I don't exactly remember but it was probably accidental.  SlowBuffer was an implementation detail - it was the backing store for normal buffers - and not something users normally interacted with directly.